src/rgw/rgw_rados.cc

```diff
@@ -8741,6 +8741,8 @@ int RGWRados::get_obj_state_impl(RGWObjectCtx *rctx, const RGWBucketInfo& bucket
       s->manifest.set_head(bucket_info.placement_rule, obj, s->size); /* patch manifest to reflect the head we just read, some manifests might be
                                              broken due to old bugs */
       s->size = s->manifest.get_obj_size();
+      if (s->accounted_size < s->size)
```

Comment:
Why do we need the check? Shouldn't we always report the `accounted_size` as the result of `get_obj_size()`?